### PR TITLE
Confluence: Windows support for finding attachments to upload

### PR DESCRIPTION
**Pull request by ghost**
## Description
Windows tests and changes for extracting attachment paths.

### CODE REVIEW

File: tests/unit/confluence.test.ts [hunk 1]
```diff
@@ -3251,5 +3337,39 @@ if (RUN_ALL_TESTS) {
   runUpdateImagePathsForContentBody();
   runCapFirstLetter();
 } else {
-  runCapFirstLetter();
+  runFindAttachments();
 }
+
+// WINDOWS
+// publishFiles: {
+//   baseDir: "C:\\Users\\Megaport\\Documents\\dev\\quarto-confluence-test-main\\simple-site2\\_site",
```
Comment (allenmanning) [inline tests/unit/confluence.test.ts:3345]: Oh no, the hackers are going to get my gaming PC now.

File: src/publish/confluence/confluence-helper.ts [hunk 2]
```diff
@@ -614,8 +622,10 @@ export const updateImagePaths = (body: ContentBody): ContentBody => {
 export const findAttachments = (
   bodyValue: string,
   publishFiles: string[] = [],
-  filePath: string = ""
+  filePathParam: string = ""
 ): string[] => {
+  const filePath = filePathParam.replaceAll("\\", "/");
```
Comment (allenmanning) [inline src/publish/confluence/confluence-helper.ts:627]: I tried Deno's path attributes and ran into some issues so I'm now adding tests and doing string replacement.  I'm not super happy with this, I want it to be all Deno.

File: src/publish/confluence/confluence-helper.ts [hunk 3]
```diff
@@ -624,17 +634,22 @@ export const findAttachments = (
 
   if (publishFiles.length > 0) {
     uniqueResult = uniqueResult.map((assetFileName: string) => {
-      const assetInPublishFiles = publishFiles.find((assetPath) => {
-        return assetPath.endsWith(`${parentPath}/${assetFileName}`);
+      const assetInPublishFiles = publishFiles.find((assetPathParam) => {
+        const assetPath = assetPathParam.replaceAll("\\", "/");
```
Comment (dragonstyle) [inline src/publish/confluence/confluence-helper.ts:638]: We do have a function which essentially does the same thing which probably makes sense to use:

`pathWithForwardSlashes`

File: src/publish/confluence/confluence-helper.ts [hunk 4]
```diff
@@ -673,3 +689,23 @@ export const getAttachmentsDirectory = (
 
   return result;
 };
+
+export const getAttachmentsDirectory = (
```
Comment (dragonstyle) [inline src/publish/confluence/confluence-helper.ts]: Is this function actually used somewhere (didn't see it but it looks a little like a work in progress)

**Review by dragonstyle — APPROVED** [review]

File: src/publish/confluence/confluence-helper.ts (hunk 4 above)
Comment (allenmanning) [reply src/publish/confluence/confluence-helper.ts]: You zoned right in on it.  Yep it has been removed- dead code.

File: src/publish/confluence/confluence-helper.ts (hunk 3 above)
Comment (allenmanning) [reply src/publish/confluence/confluence-helper.ts:638]: oo thank you

Comment (allenmanning) [reply src/publish/confluence/confluence-helper.ts]: replaced

**Pull request merged**